dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task201014UpdateColumnsValuesInIdentifierTable.java

```diff
@@ -1,102 +1,166 @@
-
-    private static final String UPDATE_CONTAINERS = "update identifier as ident, "
-            + "(select identifier, owner i_owner, idate create_date from dot_containers, inode where inode.inode=dot_containers.inode) as q "
-            + "set ident.owner=q.i_owner, ident.create_date=q.create_date "
-            + "where ident.id=q.identifier";
-
-    private static final String UPDATE_CONTAINERS_POSTGRESQL =
-            "update identifier set owner=q.i_owner, create_date=q.create_date from "
-                    + "(select identifier, owner i_owner, idate create_date from dot_containers, inode "
-                    + "where inode.inode=dot_containers.inode) q where id=q.identifier";
-
-    private static final String UPDATE_TEMPLATES = "update identifier as ident, "
-            + "(select identifier, owner i_owner, idate create_date from template, inode where inode.inode=template.inode) as q "
-            + "set ident.owner=q.i_owner, ident.create_date=q.create_date "
-            + "where ident.id=q.identifier";
-
-    protected static final String UPDATE_TEMPLATES_POSTGRESQL =
-            "update identifier set owner=q.i_owner, create_date=q.create_date from "
-                    + "(select identifier, owner i_owner, idate create_date from template, inode "
-                    + "where inode.inode=template.inode) q where id=q.identifier";
-
-    private static final String UPDATE_LINKS = "update identifier as ident, "
-            + "(select identifier, owner i_owner, idate create_date from links, inode where inode.inode=links.inode) as q "
-            + "set ident.owner=q.i_owner, ident.create_date=q.create_date "
-            + "where ident.id=q.identifier";
-
-    protected final static String UPDATE_LINKS_POSTGRESQL =
-            "update identifier set owner=q.i_owner, create_date=q.create_date from "
-                    + "(select identifier, owner i_owner, idate create_date from links, inode "
-                    + "where inode.inode=links.inode) q where id=q.identifier";
+public class Task201014UpdateColumnsValuesInIdentifierTable extends AbstractJDBCStartupTask {
 
     @Override
     public boolean forceRun() {
         return true;
     }
 
     @Override
-    public void executeUpgrade() throws DotDataException, DotRuntimeException {
+    public String getPostgresScript() {
 
-        final DotConnect dotConnect = new DotConnect()
-                .setSQL(DbConnectionFactory.isPostgres() ? UPDATE_CONTENTLETS_POSTGRESQL
-                        : UPDATE_CONTENTLETS);
-        dotConnect.loadResult();
+        final StringBuilder query = new StringBuilder();
 
-        // update owner, create_date and asset_subtype for folders
-        dotConnect.setSQL(DbConnectionFactory.isPostgres() ? UPDATE_FOLDER_POSTGRESQL
-                : UPDATE_FOLDER);
-        dotConnect.loadResult();
+        //update templates
+        query.append(getQueryToUpdateNonContentletsPostgres("template"));
```

Comment:
Codacy found an issue: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6393742)